From sandeen@sandeen.net Thu Aug 14 06:25:45 2008
From: Eric Sandeen <sandeen@sandeen.net>
Date: Thu, 14 Aug 2008 08:25:40 -0500
Subject: USB: cdc-acm: quirk for Conexant CX93010 USB modem
To: linux-usb@vger.kernel.org
Cc: oliver@neukum.name, Greg KH <greg@kroah.com>
Message-ID: <48A43254.6070400@sandeen.net>


This patch gets my Rosewill RNX-56USB USB modem (with Conexant CX93010
chipset) up and running to the point where I can send AT commands and
retrieve caller ID data, which is all I want to do with it.

Signed-off-by: Eric Sandeen <sandeen@sandeen.net>
Acked-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-acm.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -1362,6 +1362,9 @@ static struct usb_device_id acm_ids[] = 
 	{ USB_DEVICE(0x0803, 0x3095), /* Zoom Telephonics Model 3095F USB MODEM */
 	.driver_info = NO_UNION_NORMAL, /* has no union descriptor */
 	},
+	{ USB_DEVICE(0x0572, 0x1321), /* Conexant USB MODEM CX93010 */
+	.driver_info = NO_UNION_NORMAL, /* has no union descriptor */
+	},
 
 	/* control interfaces with various AT-command sets */
 	{ USB_INTERFACE_INFO(USB_CLASS_COMM, USB_CDC_SUBCLASS_ACM,
